PR for SRVLOGIC-353: Document "Managing persistence" section in the OSL docs.

kaldesai opened the pull request:
Affecting version: `serverless-docs-1.35`

Tracking JIRAs: https://issues.redhat.com/browse/SRVLOGIC-353

Doc previews: [Managing workflow persistence](https://85633--ocpdocs-pr.netlify.app/openshift-serverless/latest/serverless-logic/serverless-logic-managing-persistence)

wmedvede commented:
No that we have a "Workflow eventing system" menu entry, I think we can maybe rename en entry link to this section to "Workflow persistence"

![image](https://github.com/user-attachments/assets/945c1b8f-5b62-44a1-b78a-297708ef42d6)

wmedvede reviewed (changes requested):
Need to continue on this PR

wmedvede commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,87 @@
+
+* By default, each workflow uses a schema name equal to the workflow name. Ensure that you manually apply the schema initialization for each workflow.
+
+* If you are using the `SonataFlow` CR persistence configuration, you can specify a custom schema name.
```
```suggestion
* If you are using the `SonataFlow` CR persistence configuration, you can specify a custom schema name.

To execute the DDL scripts you must follow this procedure:

* Download the DDL scripts from the following location: link:{https://maven.repository.redhat.com/ga/org/kie/kogito/kogito-ddl/9.102.0.redhat-00005/kogito-ddl-9.102.0.redhat-00005-db-scripts.zip}[kogito-ddl-9.102.0.redhat-00005-db-scripts.zip]

* Extract the files

* Execute the .sql files present in the root directory on the target Postgresql database. You must execute the files following the given file version numbers.

For example:

. V1.35.0__create_runtime_PostgreSQL.sql
. V10.0.0__add_business_key_PostgreSQL.sql
. V10.0.1__alter_correlation_PostgreSQL.sql
. etc

[NOTE]
====
The file version numbers are not related with the {ServerlessLogicOperatorName} .
====

```

wmedvede replied:
@kaldesai Please check that this last adding is well formatted. I can't see the final layout.

For this link:     {https://maven.repository.redhat.com/ga/org/kie/kogito/kogito-ddl/9.102.0.redhat-00005/kogito-ddl-9.102.0.redhat-00005-db-scripts.zip}[kogito-ddl-9.102.0.redhat-00005-db-scripts.zip]

we need also a variable, since the link changes from version to version.
Many thanks!

kaldesai replied:
Added the content

kaldesai commented on modules/serverless-logic-profile-persistence-requirements.adoc:
```diff
@@ -0,0 +1,42 @@
+
+|`io.quarkus`
+|`quarkus-agroal`
+|`3.8.4.redhat-00002`
```
I have added the attribute for this.

kaldesai commented on modules/serverless-logic-profile-persistence-requirements.adoc:
```diff
@@ -0,0 +1,42 @@
+
+|`io.quarkus`
+|`quarkus-jdbc-postgresql`
+|`3.8.4.redhat-00002`
```
Done. I have added the attribute for this.

kaldesai commented on modules/serverless-logic-profile-persistence-requirements.adoc:
```diff
@@ -0,0 +1,42 @@
+
+|`org.kie`
+|`kie-addons-quarkus-persistence-jdbc`
+|`9.100.0.redhat-00004`
```
Done. I have added the attribute for this.

kaldesai commented on modules/serverless-logic-profile-persistence-requirements.adoc:
```diff
@@ -0,0 +1,42 @@
+
+|===
+
+If you are using the `kogito-swf-builder` to generate your images, you can pass the following build argument to include these extensions:
```
Done. I have added the attribute for this.

kaldesai commented on modules/serverless-logic-profile-persistence-requirements.adoc:
```diff
@@ -0,0 +1,42 @@
+
+[source,terminal]
+----
+$ QUARKUS_EXTENSIONS=io.quarkus:quarkus-agroal:3.8.4.redhat-00002,io.quarkus:quarkus-jdbc-postgresql:3.8.4.redhat-00002,org.kie:kie-addons-quarkus-persistence-jdbc:9.100.0.redhat-00004
```
Done!

wmedvede replied:
@kaldesai I can see that you added the variables, great work. In the table for example, the values are good. but in this particular part, In the generated document I can see  this:

![image](https://github.com/user-attachments/assets/a8b2fbff-4559-4908-8fe9-63579f037c38)

Not sure if this will be good when we merge, or must be fixed now ¿?

kaldesai commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,87 @@
+[id="serverless-logic-flyway-config-environment-variables_{context}"]
+== Flyway configuration using environment variables in the Workflow Container
+
+You can enable Flyway by adding an environment variable to the `spec.podTemplate.container` field in the `SonataFlow` CR bu using the following example:
```
Good catch! :)

kaldesai commented:
> No that we have a "Workflow eventing system" menu entry, I think we can maybe rename en entry link to this section to "Workflow persistence"
> 
> ![image](https://private-user-images.githubusercontent.com/2431454/392078418-945c1b8f-5b62-44a1-b78a-297708ef42d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5ODMxMzUsIm5iZiI6MTczMzk4MjgzNSwicGF0aCI6Ii8yNDMxNDU0LzM5MjA3ODQxOC05NDVjMWI4Zi01YjYyLTQ0YTEtYjc4YS0yOTc3MDhlZjQyZDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTJUMDU1MzU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDFjYjVkYjgwOWVmMTg3ZGJjMmIwOWMwNTFjZTY3M2Q4ZjFmZjcxZDIwMzI3MmMyYjQ5MmYwYjllN2EzZWJiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4rR6x5zjh_GCHqilkI9qNpDAdE6Uy0wD3GDkctArsDg)

Hello @wmedvede, "Workflow persistence" seems incomplete, I can make it to "Managing Workflow persistence". WDYT?

kaldesai commented:
@domhanak Can you please QE review this PR? 

@wmedvede I have incorporated all of your changes. Thank you!

wmedvede commented:
> > No that we have a "Workflow eventing system" menu entry, I think we can maybe rename en entry link to this section to "Workflow persistence"
> > ![image](https://private-user-images.githubusercontent.com/2431454/392078418-945c1b8f-5b62-44a1-b78a-297708ef42d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5ODMxMzUsIm5iZiI6MTczMzk4MjgzNSwicGF0aCI6Ii8yNDMxNDU0LzM5MjA3ODQxOC05NDVjMWI4Zi01YjYyLTQ0YTEtYjc4YS0yOTc3MDhlZjQyZDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTJUMDU1MzU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDFjYjVkYjgwOWVmMTg3ZGJjMmIwOWMwNTFjZTY3M2Q4ZjFmZjcxZDIwMzI3MmMyYjQ5MmYwYjllN2EzZWJiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4rR6x5zjh_GCHqilkI9qNpDAdE6Uy0wD3GDkctArsDg)
> 
> Hello @wmedvede, "Workflow persistence" seems incomplete, I can make it to "Managing Workflow persistence". WDYT?

+1000, great idea!

wmedvede reviewed (changes requested):
Many thanks @kaldesai for incorporating all the suggestions.
Just a small nitpick, and a comment regarding:

![image](https://github.com/user-attachments/assets/e73ecc8c-bdc0-4476-b278-ffe60106e539)


take a look please

domhanak commented:
@gmunozfe
please check

gmunozfe reviewed (approved):
Looks good to me, good work @kaldesai !

kaldesai commented:
/label peer-review-needed

kaldesai commented:
/label merge-review-needed

skopacz1 commented on modules/serverless-logic-config-persistence-using-sonataflowplaform-cr.adoc:
```diff
@@ -0,0 +1,88 @@
+<6> Key in the Secret that contains the database username.
+<7> Key in the Secret that contains the database password.
+
+When this persistence configuration is in place, the {ServerlessLogicOperatorName} configures every workflow deployed in this namespace using the `preview` or `gitops` profile, to connect with the PostgreSQL database by injecting relevant JDBC connection parameters as environment variables. 
```
This is not something that really needs to be addressed in this PR, however in this module, this is a lot of conceptual information to add after a procedure's steps. I feel like the spirit of procedural modules are to focus on instructions for a task, and talking about conceptual content in depth should be saved for concept modules.

I'd suggest considering a structure like this for the future:

* Conceptual module that introduces the topic
  * Procedural module that just focuses on the steps

skopacz1 commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,104 @@
+----
+
+[id="serverless-logic-manual-db-initialization-using-ddl_{context}"]
+== Manual database initialization using DDL scripts
```
This section has a procedure and should not be contained within a reference module, this should be pulled out into its own procedure module. It should also have the following characteristics:

- A heading that has a gerund in it (example: **Initializing** a manual database using DDL scripts)
- a .Procedure title right above the steps

kaldesai replied:
Done!

skopacz1 commented on modules/serverless-logic-persistence-precedence-rules.adoc:
```diff
@@ -0,0 +1,18 @@
+// * serverless-logic/serverless-logic-managing-persistence
+
+
+:_mod-docs-content-type: REFERENCE
```
In my opinion this is more of a conceptual module because it explains concepts in depth, whereas a reference module is more like something with information that users can quickly reference if needed, like parameters and such. See the mod docs guidelines definitions for [concept modules](https://redhat-documentation.github.io/modular-docs/#concept-module-definition) and [reference modules](https://redhat-documentation.github.io/modular-docs/#reference-module-definition).
```suggestion
:_mod-docs-content-type: CONCEPT
```

kaldesai replied:
Done

skopacz1 commented on modules/serverless-logic-profile-persistence-requirements.adoc:
```diff
@@ -0,0 +1,42 @@
+
+The persistence configurations provided for both `SonataFlowPlatform` CR and `SonataFlow` CR apply equally to the `preview` and `gitops` profiles. However, these configurations should not be used with the `dev` profile, as this profile ignores them entirely.
+
+The primary difference between the `preview` and `gitops` profiles lies in the build process:
```
I think since you're not introducing a format element like a list here, you could just end this sentence with a period. up to you:
```suggestion
The primary difference between the `preview` and `gitops` profiles lies in the build process.
```

skopacz1 reviewed (commented):
Left some comments that should be addressed before this is ready to be merged. Unfortunately I will be out on recharge, so add this back to the queue when you've addressed feedback.

kaldesai commented on modules/serverless-logic-config-persistence-using-sonataflow-cr.adoc:
```diff
@@ -0,0 +1,56 @@
+[id="serverless-logic-config-persistence-using-sonataflow-cr_{context}"]
+= Configuring persistence using the `SonataFlow` CR
+
+The `SonataFlow` CR enables workflow-specific persistence configuration. You can use this configuration independently, even if `SonataFlowPlatform` persistence is already set up in the current namespace.
```
Done

kaldesai commented on modules/serverless-logic-config-persistence-using-sonataflowplaform-cr.adoc:
```diff
@@ -0,0 +1,88 @@
+====
+
+You can configure persistence in the `SonataFlowPlatform` CR by using the following procedure:
+
```
Done!

kaldesai commented:
Note: I assigned this PR for the merge review before the recharge. The merge reviewer left some comments and I incorporated those suggestions. He was out on recharge, so I added this PR back to the queue to merge.

kaldesai commented:
/label merge-review-needed

xenolinux commented on serverless-logic/serverless-logic-managing-persistence.adoc:
```diff
@@ -0,0 +1,19 @@
+
+toc::[]
+
+This section describes how to configure a SonataFlow instance to use persistence and store workflow context in a relational database.
```
```suggestion
You can configure a `SonataFlow` instance to use persistence and store workflow context in a relational database.
```

https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc
> Avoid self-referential language, such as "This topic covers…​" or "Use this procedure to…​".

xenolinux commented on serverless-logic/serverless-logic-managing-persistence.adoc:
```diff
@@ -0,0 +1,19 @@
+
+By design, Kubernetes pods are stateless. This behavior can pose challenges for workloads that need to maintain the application state across pod restarts. In the case of {ServerlessLogicProductName}, the workflow context is lost when the pod restarts by default.
+
+To ensure workflow recovery in such scenarios, you must configure workflow runtime persistence. Use the `SonataFlowPlatform` custom resource (CR) or the `SonataFlow` custom resource (CR) to provide this configuration. The scope of the configuration varies depending on which resource you use.
```
```suggestion
To ensure workflow recovery in such scenarios, you must configure workflow runtime persistence. Use the `SonataFlowPlatform` custom resource (CR) or the `SonataFlow` CR to provide this configuration. The scope of the configuration varies depending on which resource you use.
```

Only an acronym is needed after the first occurrence.

xenolinux commented on modules/serverless-logic-config-persistence-using-sonataflow-cr.adoc:
```diff
@@ -0,0 +1,58 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="serverless-logic-config-persistence-using-sonataflow-cr_{context}"]
+= Configuring persistence using the `SonataFlow` CR
```
```suggestion
= Configuring persistence using the SonataFlow CR
```

https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references
> Do not use backticks or other markup in assembly or module headings.

xenolinux commented on modules/serverless-logic-config-persistence-using-sonataflow-cr.adoc:
```diff
@@ -0,0 +1,58 @@
+<4> Optional: Name of the database schema for workflow data. Defaults to the workflow name.
+<5> Optional: Port to connect to the PostgreSQL Service. Defaults to `5432`.
+<6> Name of the Kubernetes Secret containing database credentials.
+<7> Key in the Secret containing the database username.
+<8> Key in the Secret containing the database password.
```
```suggestion
<7> Key in the `Secret` object containing the database username.
<8> Key in the `Secret` object containing the database password.
```

xenolinux commented on modules/serverless-logic-config-persistence-using-sonataflowplaform-cr.adoc:
```diff
@@ -0,0 +1,88 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="serverless-logic-config-persistence-using-sonataflowplaform-cr_{context}"]
+= Configuring persistence using the `SonataFlowPlatform` CR
```
```suggestion
= Configuring persistence using the SonataFlowPlatform CR
```
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references
> Do not use backticks or other markup in assembly or module headings.

xenolinux commented on modules/serverless-logic-config-persistence-using-sonataflowplaform-cr.adoc:
```diff
@@ -0,0 +1,88 @@
+[id="serverless-logic-config-persistence-using-sonataflowplaform-cr_{context}"]
+= Configuring persistence using the `SonataFlowPlatform` CR
+
+The `SonataFlowPlatform` custom resource (CR) enables persistence configuration at the namespace level. This approach applies the persistence settings automatically to all workflows deployed in the namespace. It simplifies resource configuration, especially when multiple workflows in the namespace belong to the same application. While this configuration is applied by default, individual workflows ...
```
```suggestion
The `SonataFlowPlatform` custom resource (CR) enables persistence configuration at the namespace level. This approach applies the persistence settings automatically to all workflows deployed in the namespace. It simplifies resource configuration, especially when multiple workflows in the namespace belong to the same application. While this configuration is applied by default, individual workflows in the namespace can override it using the `SonataFlow` CR.
```

- Add missing back ticks

xenolinux commented on modules/serverless-logic-config-persistence-using-sonataflowplaform-cr.adoc:
```diff
@@ -0,0 +1,88 @@
+<3> Name of the PostgreSQL database for storing workflow data.
+<4> Optional: Port number to connect to the PostgreSQL service. Defaults to `5432`.
+<5> Name of the Kubernetes Secret containing database credentials.
+<6> Key in the Secret that contains the database username.
+<7> Key in the Secret that contains the database password.
```
```suggestion
<6> Key in the `Secret` object that contains the database username.
<7> Key in the `Secret` object that contains the database password.
```

xenolinux commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,104 @@
+[id="serverless-logic-persistence-db-schema-initialization_{context}"]
+= Database schema initialization
+
+When you are using `SonataFlow` with PostgreSQL persistence, you can initialize the database schema either by enabling Flyway or by manually applying database schema updates using DDL (Data Definition Language) scripts.
```
```suggestion
When you are using `SonataFlow` with PostgreSQL persistence, you can initialize the database schema either by enabling Flyway or by manually applying database schema updates using Data Definition Language (DDL) scripts.
```

xenolinux commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,104 @@
+
+When you are using `SonataFlow` with PostgreSQL persistence, you can initialize the database schema either by enabling Flyway or by manually applying database schema updates using DDL (Data Definition Language) scripts.
+
+Flyway is managed by `kie-addons-quarkus-flyway` runtime module and it is disabled by default. To enable Flyway, you must configure it using one of the following methods:
```
```suggestion
Flyway is managed by the `kie-addons-quarkus-flyway` runtime module and it is disabled by default. To enable Flyway, you must configure it using one of the following methods:
```

xenolinux commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,104 @@
+[id="serverless-logic-flyway-config-workflow-configmap_{context}"]
+== Flyway configuration in the workflow ConfigMap
+
+To enable Flyway in the workflow ConfigMap, you can add the following property:
```
```suggestion
To enable Flyway in the workflow `ConfigMap`, you can add the following property:
```

https://redhat-documentation.github.io/supplementary-style-guide/#config-map

xenolinux commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,104 @@
+
+To enable Flyway in the workflow ConfigMap, you can add the following property:
+
+.Example of enabling Flyway in the workflow ConfigMap
```
```suggestion
.Example of enabling Flyway in the workflow `ConfigMap`
```

Later on line no 33, workflow is capitalized as "Workflow" and line no 37 mentions it as "workflow" 
- Suggestion to consider using _workflow vs Workflow_ consistently.

xenolinux commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,104 @@
+----
+
+[id="serverless-logic-flyway-config-environment-variables_{context}"]
+== Flyway configuration using environment variables in the Workflow Container
```
```suggestion
== Flyway configuration using environment variables in the Workflow container
```

Suggestion to consider using workflow vs Workflow consistently.

xenolinux commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,104 @@
+
+You can enable Flyway by adding an environment variable to the `spec.podTemplate.container` field in the `SonataFlow` CR by using the following example:
+
+.Example of enabling Flyway by using the workflow container environment variable
```
```suggestion
.Example of enabling Flyway by using the workflow container environment variable
```

Suggestion to consider using workflow vs Workflow consistently.

xenolinux commented on modules/serverless-logic-persistence-precedence-rules.adoc:
```diff
@@ -0,0 +1,18 @@
+[id="serverless-logic-persistence-precedence-rules_{context}"]
+= Persistence configuration precedence rules
+
+You can use `SonataFlow` custom resource (CR) persistence independently or alongside `SonataFlowPlatform` custom resource (CR) persistence. If a `SonataFlowPlatform` CR persistence configuration exists in the current namespace, the following rules determine which persistence configuration applies:
```
```suggestion
You can use `SonataFlow` custom resource (CR) persistence independently or alongside `SonataFlowPlatform` CR persistence. If a `SonataFlowPlatform` CR persistence configuration exists in the current namespace, the following rules determine which persistence configuration applies:
```

xenolinux commented on modules/serverless-logic-persistence-precedence-rules.adoc:
```diff
@@ -0,0 +1,18 @@
+
+2. If the `SonataFlow` CR does not include a persistence configuration and the `spec.persistence` field is absent, the {ServerlessLogicOperatorName} uses the persistence configuration from the current `SonataFlowPlatform` if any.
+
+3. To disable persistence for the workflow, explicitly set `spec.persistence: {}` in the `SonataFlow` CR. This configuration ensures the workflow does not inherit persistence settings from the `SonataFlowPlatform`.
```
```suggestion
3. To disable persistence for the workflow, explicitly set `spec.persistence: {}` in the `SonataFlow` CR. This configuration ensures the workflow does not inherit persistence settings from the `SonataFlowPlatform` CR.
```

xenolinux commented on modules/serverless-logic-profile-persistence-requirements.adoc:
```diff
@@ -0,0 +1,42 @@
+[id="serverless-logic-profile-persistence-requirements_{context}"]
+= Profile specific persistence requirements
+
+The persistence configurations provided for both `SonataFlowPlatform` custom resource (CR) and `SonataFlow` custom resource (CR) apply equally to the `preview` and `gitops` profiles. However, these configurations should not be used with the `dev` profile, as this profile ignores them entirely.
```
```suggestion
The persistence configurations provided for both `SonataFlowPlatform` custom resource (CR) and `SonataFlow` CR apply equally to the `preview` and `gitops` profiles. However, you must avoid using these configurations with the `dev` profile, as this profile ignores them entirely.
```

xenolinux reviewed (commented):
Before merging; a few suggestions to consider.

xenolinux commented on modules/serverless-logic-persistence-db-schema-initialization.adoc:
```diff
@@ -0,0 +1,104 @@
+
+.Procedure
+
+. Download the DDL scripts from the {ServerlessLogic-DDL-Script-url} location.
```
This URL gives the error: Path not found in group repository "maven.repository.redhat.com root repository" [id=root_repository].

Is this expected?

kaldesai replied:
Yes, this is expected. We have a GA next week. Once we are live it will be a working URL.

xenolinux commented on modules/serverless-logic-config-persistence-using-sonataflowplaform-cr.adoc:
```diff
@@ -0,0 +1,88 @@
+PostgreSQL is currently the only supported database for persistence.
+====
+
+The following example shows the environment variables generated for a workflow named `example-workflow` deployed with the earlier `SonataFlowPlatform` configuration. These configurations specifically relate to persistence and are managed by the {ServerlessLogicOperatorName}. These settings are immutable once applied.
+
+[NOTE]
+====
+When you use the `SonataFlowPlatform` persistence, every workflow is configured to use a PostgreSQL schema name equal to the workflow name.
+====
+
+[source,yaml]
+----
+env:
+  - name: QUARKUS_DATASOURCE_USERNAME
+    valueFrom:
+      secretKeyRef:
+        name: postgres-secrets-example
+        key: POSTGRESQL_USER
+  - name: QUARKUS_DATASOURCE_PASSWORD
+    valueFrom:
+      secretKeyRef:
+        name: postgres-secrets-example
+        key: POSTGRESQL_PASSWORD
+  - name: QUARKUS_DATASOURCE_DB_KIND
+    value: postgresql
+  - name: QUARKUS_DATASOURCE_JDBC_URL
+    value: >-
+      jdbc:postgresql://postgres-example.postgres-example-namespace:1234/example-database?currentSchema=example-workflow
+  - name: KOGITO_PERSISTENCE_TYPE
+    value: jdbc
+----
```
Can this part be the separate step? Preview looks off.

kaldesai replied:
I have converted this into a 3rd step now.

xenolinux reviewed (commented):
LGTM Merging...

xenolinux commented:
/cherrypick serverless-docs-1.35
